packages/babel-types/src/definitions/es2015.js

```diff
@@ -13,7 +13,7 @@ import {
 } from "./core";
 
 defineType("AssignmentPattern", {
-  visitor: ["left", "right"],
+  visitor: ["left", "right", "decorators", "typeAnnotation"],
```

Comment:
Usually visitors try to run in code order, and I _think_ in this context, `decorators` is essentially a prefix on the assignment. Same for type annotations being attached to the left item, so this should probably be

    ["decorators", "left", "typeAnnotation", "right"],